### PR TITLE
PMREMGenerator: Refactoring

### PR DESCRIPTION
**Pull request by WestLangley**
/ping @elalish review requested

### CODE REVIEW

File: src/extras/PMREMGenerator.js [hunk 1]
```diff
@@ -136,11 +136,7 @@ PMREMGenerator.prototype = {
 	 */
 	fromEquirectangular: function ( equirectangular ) {
 
-		equirectangular.magFilter = NearestFilter;
-		equirectangular.minFilter = NearestFilter;
-		equirectangular.generateMipmaps = false;
```
Comment (elalish) [inline src/extras/PMREMGenerator.js:141]: Indeed, this looks vestigial.

File: src/extras/PMREMGenerator.js [hunk 2]
```diff
@@ -223,22 +213,34 @@ PMREMGenerator.prototype = {
...
+			encoding: _isLDR( texture ) ? texture.encoding : RGBEEncoding,
 			depthBuffer: false,
 			stencilBuffer: false
 		};
 
 		const cubeUVRenderTarget = _createRenderTarget( params );
-		cubeUVRenderTarget.depthBuffer = equirectangular ? false : true;
-		this._pingPongRenderTarget = _createRenderTarget( params );
+		cubeUVRenderTarget.depthBuffer = texture ? false : true;
+		this._pingPongRenderTarget = _createRenderTarget( params ); // what should be the pingPong depthBuffer setting?
```
Comment (elalish) [inline src/extras/PMREMGenerator.js:243]: The only thing we need a depth buffer for is when a scene is rendered into a cubemap. The blurring does not require one, so the pingPong will never need it.

**Review by elalish — APPROVED** [review]
Thanks, looks good to me.

File: src/extras/PMREMGenerator.js (hunk 2 above)
Comment (WestLangley) [reply src/extras/PMREMGenerator.js:243]: Oh, I see. You set `false` in the `params`. I missed that.

**Comment by WestLangley** [comment]
@mrdoob This should be good-to-go.

**Comment by WestLangley** [comment]
I give up on CI.   :/

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**